tests/common/include/tests/common/tools_navigator.inl

```diff
@@ -66,23 +67,26 @@ inline void check_volume_switch(state_t &state, dindex vol_id) {
 }
 
 /** Checks an entire step to next barrel surface */
-template <typename navigator_t, typename state_t = typename navigator_t::state,
-          typename track_t>
-inline void check_step(navigator_t &nav, state_t &state, track_t &trck,
-                       dindex vol_id, std::size_t n_candidates,
-                       dindex current_id, dindex next_id) {
+template <typename navigator_t, typename state_t, typename stepper_state_t>
+inline void check_step(navigator_t &nav, state_t &state,
+                       stepper_state_t &stepping, dindex vol_id,
+                       std::size_t n_candidates, dindex current_id,
+                       dindex next_id) {
+
+    auto &trck = stepping();
```

Comment:
Is this still needed? The track is advanced by the stepper now, right?

Reply:
Right.  I changed `check_steps` to take stepper and use step function instead